pkg/operator/sync.go

```diff
@@ -341,39 +347,90 @@ func (optr *Operator) syncMachineConfigController(config *renderConfig) error {
 		}
 	}
 
-	for _, path := range []string{
-		"manifests/machineconfigcontroller/events-rolebinding-default.yaml",
-		"manifests/machineconfigcontroller/events-rolebinding-target.yaml",
-	} {
+	for _, path := range paths.roleBindings {
+		rbBytes, err := renderAsset(config, path)
+		if err != nil {
+			return err
+		}
+		rb := resourceread.ReadRoleBindingV1OrDie(rbBytes)
+		_, _, err = resourceapply.ApplyRoleBinding(optr.kubeClient.RbacV1(), rb)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, path := range paths.clusterRoleBindings {
 		crbBytes, err := renderAsset(config, path)
 		if err != nil {
 			return err
 		}
-		crb := resourceread.ReadRoleBindingV1OrDie(crbBytes)
-		_, _, err = resourceapply.ApplyRoleBinding(optr.kubeClient.RbacV1(), crb)
+		crb := resourceread.ReadClusterRoleBindingV1OrDie(crbBytes)
+		_, _, err = resourceapply.ApplyClusterRoleBinding(optr.kubeClient.RbacV1(), crb)
 		if err != nil {
 			return err
 		}
 	}
 
-	crbBytes, err := renderAsset(config, "manifests/machineconfigcontroller/clusterrolebinding.yaml")
-	if err != nil {
-		return err
+	for _, path := range paths.serviceAccounts {
+		saBytes, err := renderAsset(config, path)
+		if err != nil {
+			return err
+		}
+		sa := resourceread.ReadServiceAccountV1OrDie(saBytes)
+		_, _, err = resourceapply.ApplyServiceAccount(optr.kubeClient.CoreV1(), sa)
+		if err != nil {
+			return err
+		}
 	}
-	crb := resourceread.ReadClusterRoleBindingV1OrDie(crbBytes)
-	_, _, err = resourceapply.ApplyClusterRoleBinding(optr.kubeClient.RbacV1(), crb)
-	if err != nil {
-		return err
+
+	for _, path := range paths.secrets {
+		sBytes, err := renderAsset(config, path)
+		if err != nil {
+			return err
+		}
+		s := resourceread.ReadSecretV1OrDie(sBytes)
+		_, _, err = resourceapply.ApplySecret(optr.kubeClient.CoreV1(), s)
+		if err != nil {
+			return err
+		}
 	}
 
-	saBytes, err := renderAsset(config, "manifests/machineconfigcontroller/sa.yaml")
-	if err != nil {
-		return err
+	for _, path := range paths.daemonsets {
+		dBytes, err := renderAsset(config, path)
+		if err != nil {
+			return err
+		}
+		d := resourceread.ReadDaemonSetV1OrDie(dBytes)
+		_, updated, err := resourceapply.ApplyDaemonSet(optr.kubeClient.AppsV1(), d)
+		if err != nil {
+			return err
+		}
+		if updated {
```

Comment:
In the original func this waited once and had no loops, but here we're waiting for each path? see line 559

Though this would bring up point that we only have 1 daemonset. In which case, wondering if we need array here at all?